mvds/mdf.md

```diff
@@ -0,0 +1,79 @@
+#  MVDS Meta Data Field
+
+> Version: 0.1.0 (Draft)
+> 
+> Authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+
+##  Table of Contents
+
+1. [Abstract](#abstract)
+2. [Format](#format)
+    1. [Fields](#fields) 
+1. [Usage](#usage)
+    1. [Information](#information)
+    2. [Behavior](#behavior)
+
+## Abstract
+
+In this specification, we describe a method to construct a message DAG (Directed Acyclic Graph) that will aid the consistency of [MVDS](./README.md). Additionally we explain how data sync can be used for more lightweight messages that do not require full synchronization. This specification extends the [MVDS message](./README.md#payloads) to modify the functionality of MVDS.
+
+## Format
+
+The meta data field is used to convey various information on a message and how it MUST be handled.
+
+```protobuf
+package vac.mvds;
+
+message MetaData {
+  repeated bytes parents = 7001;
+  optional int64 sequence = 7002;
+  optional bool ack = 7003 [default = true];
+}
+```
+
+We MAY transmit a `MetaData` message by extending the MVDS [message](./README.md#payloads) with a `meta_data` field.
+
+```diff
+message Message {
++ MetaData meta_data = 6000;
+  bytes group_id = 6001;
+  int64 timestamp = 6002;
+  bytes body = 6003;
+}
+```
+### Fields
+
+| Name          |  Description                                                              |
+| ------------- | ------------------------------------------------------------------------- |
+| `parents`     |  contains a list of parent [`message identifiers`](./README.md#payloads). |
+| `sequence`    |  sequence number of the message.                                          |
+| `ack`         |  contains a flag whether a message needs to be acknowledged or not.       |
+
+## Usage
+
+The flags provided through the `MetaData` message are either informational or configurational. Informational fields MAY be used by a node, configurational fields MUST be recognized and the nodes behavior MUST change accordingly.
+
+### Information
+
+Below are the list of informational flags and what they MAY be used for by a node.
+
+#### `parents`
+
+This field contains a list of a messages parents, or messages that have been seen by the node in a given context. This helps establish ordering by creating a Directed Acyclic Graph (DAG)<sup>1</sup>.
```

Comment:
This & sequence need much more explanation, without it it's difficult to make sense of the document.
What's a sequence number? Is it local to a single node or a concerted effort?
What's a context ? Group ID?
What are exactly parents? Only direct parents or ancestors as well? If ancestors is that an ordered list?
Who decides what a parent is, is that datasync (any message sent before in a group id), or the application (like in briar)?
If they are to be used for sorting, what kind of order guarantees can we make? Do we want to achieve total ordering among clients? If so how do we break ties?
What kind of consistency guarantees we make?

Reply:
Agree, it'd be useful with more detail here

Reply:
sorry for the late response @cammellos wasn't ignoring you. Sequence will get more explanation, just created an issue related to its usage in remote log [here](https://github.com/vacp2p/research/issues/1). Context is a group ID yeah. It can be parents or ancestors, @oskarth and I decided to keep it relatively open. I don't see a need for ordering, haven't thought about it that much though.

Reply:
I'll clarify those things in the spec in the next few days.